packages/core/factories/src/useLocaleFactory.ts

```diff
@@ -0,0 +1,88 @@
+import {computed, Ref, ref} from '@vue/composition-api';
+import { UseLocale } from '@vue-storefront/interfaces';
+
+export type UseLocaleFactoryParams<
+  LOCALE,
+  COUNTRY,
+  CURRENCY,
+  AVAILABLE_LOCALES = LOCALE[],
+  AVAILABLE_COUNTRIES = COUNTRY[],
+  AVAILABLE_CURRENCIES = CURRENCY[]> = {
+  loadAvailableLocales: () => Promise<AVAILABLE_LOCALES>;
+  loadAvailableCountries: () => Promise<AVAILABLE_COUNTRIES>;
+  loadAvailableCurrencies: () => Promise<AVAILABLE_CURRENCIES>;
+};
+
+export function useLocaleFactory<
+  LOCALE,
+  COUNTRY,
+  CURRENCY,
+  AVAILABLE_LOCALES = LOCALE[],
+  AVAILABLE_COUNTRIES = COUNTRY[],
+  AVAILABLE_CURRENCIES = CURRENCY[],
+  >(factoryParams: UseLocaleFactoryParams<
+  LOCALE,
+  COUNTRY,
+  CURRENCY,
+  AVAILABLE_LOCALES,
+  AVAILABLE_COUNTRIES,
+  AVAILABLE_CURRENCIES>
+) {
+  let loadingLocale = false;
+  const localeCache: Ref<LOCALE> = ref(null);
```

Comment:
can you elaborate why you called that `cache` - it doesn't seem to cache anything, perhaps using underscore to indicate it's a [private property might be better?

Reply:
@filrak I used naming "cache" to indicate that it's storing the state for computed because 'locale' is occupied by the computed itself. I am not really a fan of underscore, especially that I am not aware of using underscore naming anywhere else in this project

Reply:
Underscore is used in few places like enhancers. Imho its more descriptive than `cache` as it could be misleading @patzick @andrzejewsky wdyt?

Reply:
currentLocal, currentCountry etc etc would be great :)